File: leetcode/algorithms/292_nim_game/main.py
class NimGame:
    """
    # Complexities:
    #   - Time Complexity: O(1)
    #   - Space Complexity: O(1)
    """
    def canWinNim(self, n: int) -> bool:
        if n % 4 == 0:
            return False
        return True


    # Solution
    """
    # Complexities:
    #   - Time Complexity: O(1)
    #   - Space Complexity: O(1)
    """
    def solution(self, n: int) -> bool:
        return n % 4 != 0

File: leetcode/algorithms/292_nim_game/main.java
class NimGame {
    /**
     * Complexities:
     *   - Time Complexity: O(1)
     *   - Space Complexity: O(1)
     */
    public boolean canWinNim(int n) {
        if (n % 4 == 0) {
            return false;
        }
        return true;
    }


    // Solution
    /**
     * Complexities:
     *   - Time Complexity: O(1)
     *   - Space Complexity: O(1)
     */
    public boolean solution(int n) {
        return n % 4 != 0;
    }
}
